src/app/campaign/campaign.component.ts

```diff
@@ -101,19 +113,26 @@ export class CampaignComponent implements OnInit, OnDestroy {
   }
 
   campaignSubmit() {
-    this.campaignSaving = true;
+    this.store
+      .pipe(select(selectLocalCampaign), first())
+      .subscribe(campaign => this.store.dispatch(new actions.CampaignFormSave({ campaign })));
+
     this.campaignStoreService.storeCampaign().subscribe(([changes, deletedDocs]) => {
       this.toastr.success('Campaign saved');
-      this.campaignSaving = false;
 
       // TODO: a better way to do this. like, much better.
-      const flightId = this.router.url.split('/flight/').pop();
-      if (this.router.url.includes('/flight/') && !changes.flights[flightId]) {
-        this.router.navigate(['/campaign', changes.id]);
-      } else if (this.router.url.includes('/flight/') && flightId !== changes.flights[flightId]) {
-        this.router.navigate(['/campaign', changes.id, 'flight', changes.flights[flightId]]);
-      } else if (changes.prevId !== changes.id) {
-        this.router.navigate(['/campaign', changes.id]);
+      // TODO: when on a flight with a new campaign, don't have the new campaign.id
+      //  so only route to flight if campaign already existed,
+      //  otherwise just let campaign.effects route to the new campaign for now
```

Comment:
TODO further info: I moved the new campaign routing to campaign.effects, but I couldn't move this flight routing yet because effects doesn't handle flights yet. The campaign store no longer has information on the new campaign id because it no longer actually creates/saves the campaign, but the campaign store is still handling the flights. Previously, this was handling routing back to a new flight on a new campaign, but for now the effects routes to the campaign when you create both a flight and a campaign together. Will fix when adding flight handling to effects.